src/terminal.rs

```diff
@@ -1,32 +1,21 @@
 use crate::{
-    backend::Backend,
+    backend::{Backend, ClearType},
     buffer::Buffer,
     layout::Rect,
     widgets::{StatefulWidget, Widget},
 };
 use std::io;
 
 #[derive(Debug, Clone, PartialEq)]
-/// UNSTABLE
-enum ResizeBehavior {
-    Fixed,
-    Auto,
-}
-
-#[derive(Debug, Clone, PartialEq)]
-/// UNSTABLE
-pub struct Viewport {
-    area: Rect,
-    resize_behavior: ResizeBehavior,
+pub enum Viewport {
+    Fullscreen,
+    Inline(u16),
+    Fixed(Rect),
 }
 
 impl Viewport {
-    /// UNSTABLE
     pub fn fixed(area: Rect) -> Viewport {
-        Viewport {
-            area,
-            resize_behavior: ResizeBehavior::Fixed,
-        }
+        Self::Fixed(area)
     }
 }
```

Comment:
Let's also implement pub fn inline(u16) and fullscreen() constructors...

Reply:
@conradludgate

Reply:
Actually I'd rather delete this one. It's commented as "unstable" so it shouldn't be in use anyway, and it would be a small change of
```diff
- Viewport::fixed(n)
+ Viewport::Fixed(n)
```

Reply:
Sounds good, as long as it's consistent.